### PR TITLE
Update docs for Model-centric UI

### PR DESCRIPTION
**Pull request by edmundito**
## Change description

This change updates the docs copy and screenshots for the new model-centric UI.

TODO:
- [x] Update sources documentation
- [x] Update sample record documentation
- [x] Explain primary and secondary source, primary key in concepts
- [x] Update screenshots in other config documentation

Before merge:
- [x] Update "last updated" date for all updated pages

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

**Comment by edmundito** [comment]
I took a stab at updating the sources documentation but there's still a lot of work left to do. Please let me know what you think so far.

https://www-git-180859324-mcu-docs-grouparoo.vercel.app/docs/config/sources

File: pages/docs/config/models.mdx
Comment (evantahler) [inline pages/docs/config/models.mdx]: I don't think we need this line

File: pages/docs/config/models.mdx
Comment (evantahler) [inline pages/docs/config/models.mdx:46]: I don't think this image belongs here?

File: pages/docs/config/models.mdx
Comment (evantahler) [inline pages/docs/config/models.mdx]:
```suggestion
See [Product Concepts documentation](/docs/getting-started/product-concepts#model) to learn more.
```

File: pages/docs/config/sources.mdx
Comment (evantahler) [inline pages/docs/config/sources.mdx]:
```suggestion
A Source is a definition for how you pull data from an App into Grouparoo. [See this page](/docs/getting-started/product-concepts#source) to learn more about the core concepts in Grouparoo.
Write
```

File: pages/docs/config/sources.mdx
Comment (evantahler) [inline pages/docs/config/sources.mdx]:
```suggestion
- **Primary Source**: This is the source that provides the bulk of the data to the Model. You can only have one per Model. The Primary Source will provide the unique identifier for Records in this Model, like `user_id` or `email_address`.
```

**Review by evantahler — CHANGES_REQUESTED** [review]
Good start! Some comments below

File: pages/docs/config/models.mdx
Comment (Adorism) [inline pages/docs/config/models.mdx:28]: I think it's worth considering if it makes sense to add a table here to help distinguish between the model types. I'm especially curious if there is any further guidance we can provide as to what factors would suggest you want a custom model.

File: pages/docs/config/sources.mdx
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
Choose the App you'd like to use to feed your source. The choices shown will be based on which Apps you've already connected. If you don't see the App you want to use, you can [add a new App](/docs/config/apps).
```

File: pages/docs/config/sources.mdx
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
You will also need to indicate what makes each Record in your Source unique. This must be a unique identifier such a `user id`, `uid`, or `email`.
```

File: pages/docs/config/sources.mdx
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
When defining the Model's Primary Source, you will only need to select the identifier.
```

File: pages/docs/config/sources.mdx
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
Then, Grouparoo will generate a Property that will serve as the Primary Key for the Model where the Source is located.
```

File: pages/docs/config/sources.mdx
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
When defining a Secondary Source, you can map an attribute in the Source to an existing property that ties it back to the Primary Source.
```

File: pages/docs/config/sources.mdx
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
In the example below, the Pg Purchases Source is being tied back to the Primary Source (Pg Users) via the `user_id` foreign key in the `purchases` table.
```

File: pages/docs/config/models.mdx
Comment (edmundito) [reply pages/docs/config/models.mdx:46]: I moved the image to next steps. I just wanted to show what it looked liked after creating the model

File: pages/docs/config/models.mdx
Comment (edmundito) [reply pages/docs/config/models.mdx:28]: Afaik, there is no specific difference between the different models other than to more easily identify what type of object the user setup. Maybe I could explain this a little better here?

**Comment by edmundito** [comment]
I finalized the first pass on the documentation and added/updated everything that was needed. Let me know what you think!

File: pages/docs/config/destinations.mdx
Comment (andyjih) [inline pages/docs/config/destinations.mdx]:
```suggestion
date: "2022-01-18"
```

File: pages/docs/config/groups.mdx
Comment (andyjih) [inline pages/docs/config/groups.mdx]:
```suggestion
Groups are where the magic happens in *Group*aroo. (It's in the name, after all!) Groups are a segment or cohort of Records. They are useful in that you can use Groups to target a specific set of Records to [export to some Destination](/docs/config/destinations). [See here](/docs/getting-started/product-concepts) to learn more about the core concepts in Grouparoo.
```

File: pages/docs/config/records.mdx
Comment (andyjih) [inline pages/docs/config/records.mdx]:
```suggestion
date: "2022-01-18"
```

File: pages/docs/config/sources.mdx
Comment (andyjih) [inline pages/docs/config/sources.mdx]:
```suggestion
Select your Model in the left side navigation and click the "Create Primary Source" (or "Add new Source" for Secondary Sources) button.
```

File: pages/docs/config/sources.mdx
Comment (andyjih) [inline pages/docs/config/sources.mdx]:
```suggestion
In the example below, the "Pg Purchases" Source is being tied back to the Primary Source (Pg Users) via the `user_id` foreign key in the `purchases` table.
```

File: pages/docs/getting-started/product-concepts.mdx
Comment (andyjih) [inline pages/docs/getting-started/product-concepts.mdx]:
```suggestion
The [Mapping](#mapping) of the Primary Source is set via the Model's [Primary Key](#primary-key).
```

**Review by andyjih — CHANGES_REQUESTED** [review]
Lots of little tweaks, but overall these updates look great!

File: pages/docs/config/destinations.mdx
Comment (edmundito) [reply pages/docs/config/destinations.mdx]: I'm waiting until we're live/before merge to update these dates.

**Comment by edmundito** [comment]
I applied review feedback and did another pass on the docs to expand on the concepts or fix some of the wording. Please take another look.

**Review by evantahler — APPROVED** [review]

File: pages/docs/config/destinations.mdx
Comment (evantahler) [inline pages/docs/config/destinations.mdx]:
```suggestion
Once your Records have all the data you need via Properties and Groups, you're ready to setup your Destinations.
```

File: pages/docs/config/sources.mdx
Comment (evantahler) [inline pages/docs/config/sources.mdx:66]: Maybe there should be sections
```
#### Primary Source
#### Secondary Source
```

**Review by andyjih — APPROVED** [review]

**Pull request merged**
